src/amqp10/applicationProperties.ts

```diff
@@ -0,0 +1,55 @@
+import { MessageApplicationProperties } from "../producer"
+import { DataReader } from "../responses/raw_response"
+import { FormatCode } from "./decoder"
+
+export class ApplicationProperties {
+  public static Parse(dataReader: DataReader, elementsLength: number): MessageApplicationProperties {
+    const numEntries = elementsLength / 2
+    const messageApplicationProperties: MessageApplicationProperties = {}
+
+    for (let index = 0; index < numEntries; index++) {
```

Comment:
bonus point if you can replace this for with a reduce

Reply:
I don't really understand what you mean by reduce. You mean with a range and a map or with the array function reduce?

Reply:
range(numEntries).reduce(..., {})
@albertobarrila can help you ...

Reply:
Pay attention we don't have lodash so we dont' have `¶ange` but If remember correctly we have implemented it in some test ...